source/common/common/byte_order.h

```diff
@@ -25,3 +25,147 @@
 #include <endian.h>
 
 #endif
+
+enum class ByteOrder { Host, LittleEndian, BigEndian };
+
+template <ByteOrder, typename Integral, size_t = sizeof(Integral)> struct EndiannessConverter;
+
+// convenience function that converts an integer from host byte-order to a specified endianness
+template <ByteOrder Endianness, typename T> inline T toEndianness(T value) {
+  return EndiannessConverter<Endianness, T>::to(value);
+}
+
+// convenience function that converts an integer from a specified endianness to host byte-order
+template <ByteOrder Endianness, typename T> inline T fromEndianness(T value) {
+  return EndiannessConverter<Endianness, T>::from(value);
+}
+
+// convenience function that converts a host byte-order integer to little endian byte-order
+template <typename T> inline T toLittleEndian(T value) {
```

Comment:
it was used in the initial versions of the MySQL health checker, prior to moving this functionality to the buffer implementation

it can still prove itself to be useful, but I have no strong opinion on whether to keep it or ditch it

your call entirely

Reply:
I think if you don't have a use for it in the mysql health checker, it should be removed.

Reply:
removed